h2o-core/src/test/java/hex/AUC2Test.java

```diff
@@ -31,4 +31,30 @@ public void checkRecallValidity() throws Exception {
     auc2.checkRecallValidity();
   }
 
-}
\ No newline at end of file
+  @Test
+  /*
+   * Test the absolute MCC is calculated correctly with imprecision caused by double values
+   * PUBDEV-7426
+   */
+  public void checkAbsoluteMCC() {
+    double tp = 1005.1403389830544;
+    double tn = 4794.8034643570945;
+    double fp = 0;
+    double fn = 0;
+    double[] tps = {0, tp};
+    double[] fps = {tn, 0};
+    AUC2.AUCBuilder bldr = new AUC2.AUCBuilder(2);
+    bldr._n = 2;
+    System.arraycopy(tps, 0, bldr._tps, 0, tps.length);
+    System.arraycopy(fps, 0, bldr._fps, 0, fps.length);
+    AUC2 auc2 = new AUC2(bldr);
+    // index = 0, because AUCBuilder switch the order of array
+    double mcc = AUC2.ThresholdCriterion.absolute_mcc.exec(auc2, 0);
+    double exact_mcc = (tp*tn - fp*fn);
+    exact_mcc /= Math.sqrt((tp+fp)*(tp+fn)*(tn+fp)*(tn+fn));
+    assert mcc == 1: "It should return MCC = 1, but it returns "+ mcc;
```

Comment:
please use assertEquals

Reply:
Why do you suggest to use assertEquals? Because of the delta? The result should be exactly 1.0, nothing else. The delta should be 0 in assertEquals and, it does not make other sense to me to prefer assertEquals.

Reply:
Assertions can be disabled. In our test framework, we enable them to always be on but in your IDE you have no such guarantee. In my IntelliJ `assert 1 == 0` will pass by default.

Another big reason is reporting - if the assertion fails you want to see the value, not just that it fails.

Reply:
Ok, I will fix it! Thank you for your explanation.